src/main/webapp/WEB-INF/applicationContext.xml

```diff
@@ -2,10 +2,13 @@
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xmlns:context="http://www.springframework.org/schema/context"
        xsi:schemaLocation="http://www.springframework.org/schema/beans
-  http://www.springframework.org/schema/beans/spring-beans.xsd
-  http://www.springframework.org/schema/context
-  http://www.springframework.org/schema/context/spring-context.xsd"
->
-    <bean id="greetingService" class="org.noteskeeper.server.GreetingService"/>
+           http://www.springframework.org/schema/beans/spring-beans-2.5.xsd
```

Comment:
This are Jersey examples, not Spring ones. From Jersey perspective it is Ok to show you XML way to configure, since it still exists in the huge number of legacy applications. There is also annotation driven example - https://github.com/jersey/jersey/tree/2.25.1/examples/helloworld-spring-annotations. But it is pretty crappy. 
XML or Java configs, it is mostly related to Spring. It is not dependent on Jersey. In general I would advise switch to Spring Boot. But maybe it is better to start learning with plain Spring. @fland what do you think?
https://github.com/spring-projects/spring-boot/tree/master/spring-boot-samples/spring-boot-sample-jersey spring boot example.
Also I noted that they use <artifactId>jersey-spring4</artifactId> dependency. Maybe it is more convenient with Spring 4.
Also you may try to browse this repo (without Spring Boot) - https://github.com/peeskillet/underdog-jersey-spring-example/tree/master/src/main, it is not official, but may be handy.
Additionally, as I found, number of people are arguing that spring integration jersey library is not good at all.